components/verse/ToastNotification/ToastNotification.tsx

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect } from 'react'
+import { BsCheckCircle, BsInfoCircle, BsX, BsXCircle } from 'react-icons/bs'
+
+interface Props {
+   type: 'success' | 'info' | 'error'
+   title: string
+   visibility?: boolean
+   children: string
+}
+
+type BgColors = 'bg-[#ffffff]' | 'bg-[#b6f8c4]' | 'bg-[#b2e6f5]' | 'bg-[#ffb7b7]'
+
+const ToastNotification: React.FC<Props> = ({ type, title, visibility, children }) => {
+   const [showToast, setShowToast] = useState(visibility)
+   useEffect(() => setShowToast(visibility), [visibility])
+
+   const [backgroundHex, setBackgroundHex] = useState<BgColors>('bg-[#ffffff]')
+   useEffect(() => {
+      type === 'success' && setBackgroundHex('bg-[#b6f8c4]')
+      type === 'info' && setBackgroundHex('bg-[#b2e6f5]')
+      type === 'error' && setBackgroundHex('bg-[#ffb7b7]')
+
+      setTimeout(() => setShowToast(false), 5000)
+   }, [])
+
+   const newTitle = title.charAt(0).toUpperCase() + title.slice(1).toLowerCase()
+
+   return (
+      <div
+         className={`${
+            showToast ? '' : 'hidden'
```

Comment:
This confirms that you may use the toast more than once for the same component, because it is hidden, not unmounted. If it was unmounted then once it would be mounted again that `useEffect` without dependencies would run the `setTimeout`. I think you should keep this as is a fix the `useEffect`